ts/input/tex/Configuration.ts

```diff
@@ -278,3 +233,169 @@ export namespace ConfigurationHandler {
   };
 
 }
+
+
+/**
+ * Parser configuration combines the configurations of the currently selected
+ * packages.
+ * @constructor
+ */
+export class ParserConfiguration {
+
+  /**
+   * Priority list of init methods.
+   * @type {FunctionList}
+   */
+  protected initMethod: FunctionList = new FunctionList();
+
+  /**
+   * Priority list of init methods to call once jax is ready.
+   * @type {FunctionList}
+   */
+  protected configMethod: FunctionList = new FunctionList();
+
+  /**
+   * An ordered list of cofigurations.
+   * @type {PrioritizedList<Configuration>}
+   */
+  protected configurations: PrioritizedList<Configuration> = new PrioritizedList();
+
+  /**
+   * The subhandlers for this configuration.
+   * @type {SubHandlers}
+   */
+  public handlers: SubHandlers = new SubHandlers();
+
+  /**
+   * The collated stack items.
+   * @type {StackItemConfig}
+   */
+  public items: StackItemConfig = {};
+
+  /**
+   * The collated tag configurations.
+   * @type {TagsConfig}
+   */
+  public tags: TagsConfig = {};
+
+  /**
+   * The collated options.
+   * @type {OptionList}
+   */
+  public options: OptionList = {};
+
+  /**
+   * The collated node creators.
+   * @type {{[key: string]: any}}
+   */
+  public nodes: {[key: string]: any}  = {};
+
+
+  /**
+   * @constructor
+   */
+  constructor(packages: (string | [string, number])[]) {
+    for (const pkg of packages.slice().reverse()) {
+      this.addPackage(pkg);
+    }
+    for (let {item: config, priority: priority} of this.configurations) {
+      this.append(config, priority);
+    }
+  }
+
+  /**
+   * Initmethod for the configuration;
+   *
+   * @param {Configuration} configuration   The configuration where this one is being initialized
+   */
+  public init() {
+    this.initMethod.execute(this);
+  }
+
+  /**
+   * Init method for when the jax is ready
+   *
+   * @param {Configuration} configuration   The configuration where this one is being initialized
+   * @param {TeX} jax                       The TeX jax for this configuration
+   */
+  public config(jax: TeX<any, any, any>) {
+    this.configMethod.execute(this, jax);
+    for (const config of this.configurations) {
+      this.addFilters(jax, config.item);
+    }
+  }
+
+  /**
+   * Retrieves and adds configuration for a pacakge with priority.
+   * @param {(string | [string, number]} pkg Package with priority.
+   */
+  public addPackage(pkg: (string | [string, number])) {
+    const name = typeof pkg === 'string' ? pkg : pkg[0];
+    let conf = ConfigurationHandler.get(name);
+    if (conf) {
+      this.configurations.add(
+        conf, typeof pkg === 'string' ? conf.priority : pkg[1]);
+    }
```

Comment:
Would it be simpler to make this

```
const [name, priority] = typeof pkg === 'string' ? [pkg, config.priority] : pkg;
const conf = ConfigurationHandler.get(name);
if (conf) {
      this.configurations.add(conf, priority);
}
```

Reply:
The problem is that in the first line you do not yet have the configuration `config`. So you need the name first to look up the configuration. Then you add it either with the given priority or the default configuration if none is given. I know, it looks ugly. I reduce it to only use `typeof pkg === 'string'` once.

Reply:
Now I recall why it was like this: Typescript complains if the test is done elsewhere. So we currently avoid some casting.
So I'll leave it as is.

Reply:
Got it.  Makes sense.